pytorch/training/docker/1.8.0/py3/cu111/Dockerfile.gpu

```diff
@@ -0,0 +1,259 @@
+ENV DGLBACKEND=pytorch
+ENV CMAKE_PREFIX_PATH="$(dirname $(which conda))/../"
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_pytorch_container.training:main
+ENV MANUAL_BUILD=0
+
+ARG PT_TRAINING_URL=https://aws-pytorch-binaries.s3-us-west-2.amazonaws.com/r1.8.0_aws/20210306-000727/b5cc095ca218fb9eab43948dc83c7a020ab1db26/gpu/torch-1.8.0-cp36-cp36m-manylinux1_x86_64.whl
+ARG PT_TORCHVISION_URL=https://torchvision-build.s3-us-west-2.amazonaws.com/1.8.0/gpu/torchvision-0.9.0-cp36-cp36m-linux_x86_64.whl
+ARG SMD_MODEL_PARALLEL_URL=https://sagemaker-distributed-model-parallel.s3.amazonaws.com/pytorch-1.7/build-artifacts/2021-01-19-23-41/smdistributed_modelparallel-1.2.0-cp36-cp36m-linux_x86_64.whl
+ARG SMDATAPARALLEL_BINARY=https://smdataparallel.s3.amazonaws.com/binary/pytorch/1.8.0/cu111/2021-03-04/smdistributed_dataparallel-1.0.0-cp36-cp36m-linux_x86_64.whl
+
+RUN apt-get update \
+ && apt-get install -y --allow-change-held-packages --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    cmake \
+    cuda-command-line-tools-11-1 \
+    cuda-cudart-11-1 \
+    cuda-libraries-11-1 \
+    curl \
+    emacs \
+    git \
+    jq \
+    libcublas-11-1=${CUBLAS_VERSION}-1 \
+    libcublas-dev-11-1=${CUBLAS_VERSION}-1 \
+    libcudnn8=$CUDNN_VERSION-1+cuda11.1 \
+    libcufft-dev-11-1 \
+    libcurand-dev-11-1 \
+    libcusolver-dev-11-1 \
+    libcusparse-dev-11-1 \
+    libglib2.0-0 \
+    libgl1-mesa-glx \
+    libnccl2=${NCCL_VERSION}-1+cuda11.1 \
+    libnccl-dev=${NCCL_VERSION}-1+cuda11.1 \
+    libsm6 \
+    libxext6 \
+    libxrender-dev \
+    libgomp1 \
+    libibverbs-dev \
+    libhwloc-dev \
+    libnuma1 \
+    libnuma-dev \
+    libssl1.1 \
+    libtool \
+    openssl \
+    python3-dev \
+    vim \
+    wget \
+    zlib1g-dev \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=$OPEN_MPI_PATH \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+ENV PATH="$OPEN_MPI_PATH/bin:$PATH"
+ENV LD_LIBRARY_PATH="$OPEN_MPI_PATH/lib/:$LD_LIBRARY_PATH"
+
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
+ && curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p ${CONDA_PREFIX} \
+ && rm ~/miniconda.sh \
+ && ${CONDA_PREFIX}/bin/conda install -c conda-forge \
+    python=$PYTHON_VERSION \
+ && ${CONDA_PREFIX}/bin/conda install -y -c anaconda \
+    # conda 4.9.2 requires ruamel_yaml to be installed. Currently pinned at latest.
+    ruamel_yaml==0.15.87 \
+    numpy \
+    ipython \
+    mkl \
+    mkl-include \
+    cython \
+    typing \
+    future \
+    "pyopenssl>=17.5.0" \
+ && conda install -y -c dglteam dgl-cuda11.1==0.6.0post1 \
```

Comment:
Why is dgl pinned to a version here while it is left unpinned in the CPU DLC?

Reply:
this is dgl team's newly built binary for CUDA 11.1. They had an issue with CUDA11.1 and got a temporary fix to it. They can update by a PR later on.
however CPU didn't have an upgrade like this, so the version is unpined.

Reply:
according to DGL team we don't need to pin the version for both, modifying.